bats/tests/k8s/helm-install-rancher.bats

```diff
@@ -84,6 +87,68 @@ determine_chart_version() {
         fail || return
 }
 
+assert_not_empty_list() {
+    run "$@"
+    assert_success || return
+    run jq_output length
+    assert_success || return
+    refute_output 0 || return
+}
+
+assert_true() {
+    run --separate-stderr "$@"
+    assert_success || return
+    is_true "$output" || return
+}
+
+# Given namespace and app name, assert that a log line contains the given string.
+assert_pod_log_line() {
+    local namespace="$1"
+    local selector="app=$2"
+    shift 2
+    local expect="$*"
+    run kubectl get pod --namespace "$namespace" --selector "$selector" --output=jsonpath='{.items[0].metadata.name}'
+    assert_success
+    assert_output || return
+    local name="$output"
+
+    run kubectl logs --namespace "$namespace" "$name"
+    assert_success || return
+    assert_output --partial "$expect" || return
+}
+
+# Pull down the image manually first so we are less likely to time out when
+# deploying rancher
+pull_rancher_image() {
+    local rancher_chart_version
+    if ! load_var rancher_chart_version; then
+        fail "Could not restore Rancher chart version"
+    fi
+    if using_docker; then
+        try docker pull --quiet "rancher/rancher:v$rancher_chart_version"
+    else
+        try nerdctl pull --namespace k8s.io --quiet "rancher/rancher:v$rancher_chart_version"
+    fi
+}
+
+wait_for_rancher_pod() {
+    try --max 60 --delay 10 assert_pod_log_line cattle-system rancher Listening on :443
+    try --max 60 --delay 10 assert_pod_log_line cattle-system rancher Starting catalog controller
+    try --max 60 --delay 10 assert_pod_log_line cattle-system rancher Watching metadata for rke-machine-config.cattle.io/v1
+    try --max 60 --delay 10 assert_pod_log_line cattle-system rancher 'Creating clusterRole for roleTemplate Cluster Owner (cluster-owner).'
+    try --max 60 --delay 10 assert_pod_log_line cattle-system rancher Rancher startup complete
+    try --max 120 --delay 10 assert_pod_log_line cattle-system rancher Created machine for node
```

Comment:
I was a little shocked that the total timeouts add up to 70 minutes, but I guess there must be progress at least every 10 minutes.

Can the final step really take another 20 minutes???

Anyways, do we need a `|| return` on each `try` statement, for consistency?

I guess not, if we follow the rule that `wait_for_*` functions are **never** called via `try`. Maybe we need a linter rule for that (not in this PR).

Reply:
Yeah, in practice we never need that much. I'll go tone down the max retries, based on what actually happened in the runs.